src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -728,18 +731,21 @@ async def start_grpc_server(
 
     server = aio.server()
     service = TextGenerationService(engine, args)
+
     await service.post_init()
 
     generation_pb2_grpc.add_GenerationServiceServicer_to_server(service, server)
 
-    # TODO add reflection
+    health_servicer = health.HealthServicer()
+    health_pb2_grpc.add_HealthServicer_to_server(health_servicer, server)
```

Comment:
We've had problems doing this in the past because when we need to turn on mTLS for the server here, the kube health probe is sometimes impossible to configure when the server only accepts requests signed with a CA that we don't have access to.

But, I can't say the same isn't also just true for the REST server, anyway 🤷

Reply:
Is that the reason for https://github.com/caikit/caikit/blob/main/caikit_health_probe/__main__.py?

Reply:
Can you elaborate on the CA issue with mTLS? An example would be great

Reply:
@joerunde @prashantgupta24


Cleaned up things a little in the last push, but most importantly, I added a `grpc_healthcheck` cli that can be used as an exec probe. 

It's fairly basic, but we can extend it for more usecase as soon as we find out what's needed

Reply:
@dtrifiro re: mTLS, my basic understanding is that the server will only accepts requests if the client presents a certificate that is signed by a specific CA. Usually for our own deployments, this isn't an issue because the server and client are using certs signed by the same CA that we own. But there are some cases where a server needs to be configured to only accept requests from a customer's cluster, which will be signed by the customer's CA. We ran into problems getting health checks to work properly because we would need to generate a certificate for the client to use that is signed by the customer's CA. As far as I'm aware, this is impossible for us to do because we do not have access to the customer's CA's private key.
Specifically this would look like us setting `SSL_CA_CERTS=/some/customer/ca/cert`

For TGIS/vLLM we haven't run into this specific problem yet because these deployments usually sit behind the router + another proxy, and usually it's the proxy that has to be configured to accept requests from an external CA.

Reply:
I'll go on and merge this then, we can take this issue on if we ever encounter it. Thanks